modelscan/skip.py

```diff
@@ -0,0 +1,47 @@
+import abc
+import logging
+from enum import Enum
+from pathlib import Path
+from typing import Any, List, Union, Dict, Optional
+
+from collections import defaultdict
+
+logger = logging.getLogger("modelscan")
+
+
+class SkipCategories(Enum):
+    SCAN_NOT_SUPPORTED = 1
+    BAD_ZIP = 2
+    MODEL_CONFIG = 3
+    H5_DATA = 4
+    NOT_IMPLEMENTED = 5
+
+
+class Skip:
+    scan_name: str
+    category: SkipCategories
+    message: str
+    source: str
+
+    def __init__(self) -> None:
+        pass
+
+    def __str__(self) -> str:
+        raise NotImplementedError()
+
+
+class ModelScanSkipped:
+    def __init__(
+        self,
+        scan_name: str,
+        category: SkipCategories,
+        message: str,
+        source: str,
+    ) -> None:
+        self.scan_name = scan_name
+        self.category = category
+        self.message = message or "None"
```

Comment:
Nit: Should we just make the message an `Optional` type instead, so we don't need to hardcode `"None"`? Then we can format the `__str__` output to not include any message if it's not provided

Reply:
Thank you for the suggestion. I have removed None for `message` - it should not be optional and updated the `__str__` to show source when available